pkg/controllers/localbuild/controller.go

```diff
@@ -89,6 +104,48 @@ func (r *LocalbuildReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
+	if r.Config.DevMode {
```

Comment:
In general, I think the flow here should be:

If the static field is set to true:
1. Attempt authentication by using the static password.
1. If successful, there's nothing to do so we continue with the rest of the processing logic.
1. If fails to authenticate (status code 40X, I think), we assume the password has not changed.
1. Get the current password then set it to the static password.